packages/popover2/src/popover2.tsx

```diff
@@ -327,46 +339,58 @@ export class Popover2<T> extends AbstractPureComponent2<Popover2Props<T>, IPopov
 
         const ref = mergeRefs(popperChildRef, this.targetRef);
 
-        const targetEventHandlers = isHoverInteractionKind
-            ? {
-                  // HOVER handlers
-                  onBlur: this.handleTargetBlur,
-                  onContextMenu: this.handleTargetContextMenu,
-                  onFocus: this.handleTargetFocus,
-                  onMouseEnter: this.handleMouseEnter,
-                  onMouseLeave: this.handleMouseLeave,
-              }
-            : {
-                  // CLICK needs only one handler
-                  onClick: this.handleTargetClick,
-                  // For keyboard accessibility, trigger the same behavior as a click event upon pressing ENTER/SPACE
-                  onKeyDown: (event: React.KeyboardEvent<HTMLElement>) =>
-                      // eslint-disable-next-line deprecation/deprecation
-                      Keys.isKeyboardClick(event.keyCode) && this.handleTargetClick(event),
-              };
+        const targetEventHandlers: Popover2HoverTargetHandlers<T> | Popover2ClickTargetHandlers<T> =
+            isHoverInteractionKind
+                ? {
+                      // HOVER handlers
+                      onBlur: this.handleTargetBlur,
+                      onContextMenu: this.handleTargetContextMenu,
+                      onFocus: this.handleTargetFocus,
+                      onMouseEnter: this.handleMouseEnter,
+                      onMouseLeave: this.handleMouseLeave,
+                  }
+                : {
+                      // CLICK needs only one handler
+                      onClick: this.handleTargetClick,
+                      // For keyboard accessibility, trigger the same behavior as a click event upon pressing ENTER/SPACE
+                      onKeyDown: (event: React.KeyboardEvent<HTMLElement>) =>
+                          // eslint-disable-next-line deprecation/deprecation
+                          Keys.isKeyboardClick(event.keyCode) && this.handleTargetClick(event),
+                  };
         // Ensure target is focusable if relevant prop enabled
         const targetTabIndex = openOnTargetFocus && isHoverInteractionKind ? 0 : undefined;
-        const targetProps = {
+        const ownTargetProps = {
             "aria-haspopup":
                 this.props.popupKind ??
-                (this.props.interactionKind === Popover2InteractionKind.HOVER_TARGET_ONLY ? undefined : "true"),
+                (this.props.interactionKind === Popover2InteractionKind.HOVER_TARGET_ONLY
+                    ? undefined
+                    : ("true" as "true")),
```

Comment:
could just do "true" as const ?

Reply:
nope, unfortunately I tried that -- it only works for const enums